pci/src/bus.rs

```diff
@@ -356,8 +356,29 @@ impl BusDevice for PciConfigMmio {
     }
 }
 
+// Parse the MMIO address offset to a (bus, device, function, register) tuple.
+fn parse_mmio_config_address(config_address: u32) -> (usize, usize, usize, usize) {
+    const BUS_NUMBER_OFFSET: usize = 20;
+    const BUS_NUMBER_MASK: u32 = 0x00ff;
+    const DEVICE_NUMBER_OFFSET: usize = 15;
+    const DEVICE_NUMBER_MASK: u32 = 0x1f;
+    const FUNCTION_NUMBER_OFFSET: usize = 12;
+    const FUNCTION_NUMBER_MASK: u32 = 0x07;
+    const REGISTER_NUMBER_OFFSET: usize = 2;
+    const REGISTER_NUMBER_MASK: u32 = 0x3ff;
```

Comment:
Could you give more feedback (maybe a link to the spec) about these constants?

Reply:
https://wiki.osdev.org/PCI_Express#Enhanced_Configuration_Mechanism